File: src/main/java/net/jgp/books/spark/ch13/lab999_functions/Unbase64App.java
package net.jgp.books.spark.ch13.lab999_functions;

import static org.apache.spark.sql.functions.base64;
import static org.apache.spark.sql.functions.col;
import static org.apache.spark.sql.functions.unbase64;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;
import org.apache.spark.sql.types.DataTypes;

/**
 * unbase64 function.
 * 
 * @author jgp
 */
public class Unbase64App {

  public static void main(String[] args) {
    Unbase64App app = new Unbase64App();
    app.start();
  }

  /**
   * The processing code.
   */
  private void start() {
    SparkSession spark = SparkSession.builder()
        .appName("unbase64 function")
        .master("local[*]")
        .getOrCreate();

    Dataset<Row> df = spark.read().format("csv")
        .option("header", true)
        .load("data/functions/strings.csv");

    System.out.println("Output as array of bytes:");
    df = df
        .withColumn("base64", base64(col("fname")))
        .withColumn("unbase64", unbase64(col("base64")));
    df.show(5);

    System.out.println("Output as strings:");
    df = df
        .withColumn("name", col("unbase64").cast(DataTypes.StringType));
    df.show(5);

  }
}

File: src/main/python/lab999_functions/unbase64App.py
"""
 unbase64 function.

 @author rambabu.posa
"""
from pyspark.sql import SparkSession
from pyspark.sql import (functions as F)
from pyspark.sql.types import StringType
import os

current_dir = os.path.dirname(__file__)
relative_path = "../../../../data/functions/strings.csv"
absolute_file_path = os.path.join(current_dir, relative_path)

# Creates a session on a local master
spark = SparkSession.builder.appName("unbase64 function") \
    .master("local[*]").getOrCreate()

df = spark.read.format("csv") \
    .option("header", True) \
    .load(absolute_file_path)

print("Output as array of bytes:")

df = df.withColumn("base64", F.base64(F.col("fname"))) \
       .withColumn("unbase64", F.unbase64(F.col("base64")))

df.show(5)

print("Output as strings:")

df = df.withColumn("name", F.col("unbase64").cast(StringType()))
# Or use can use this
# df = df.withColumn("name", F.col("unbase64").cast("string"))

df.show(5)

spark.stop()